### PR TITLE
fix: logits processor state at each step

### PR DESCRIPTION
**Pull request by wallashss**
# Description

This PR fixes the update of logits processors that need to be updated at each engine step. To validate the change, I updated the existing test for min tokens where we can identify the wrong behaviour. Note: the bug is reproducible in both CB and SB.

### CODE REVIEW

File: tests/e2e/test_sampling_params.py [hunk 1]
```diff
@@ -271,8 +271,12 @@ def test_spyre_batch1_min_tokens(model: ModelInfo, backend, monkeypatch,
...
     output2 = spyre_model.generate(prompt, params2)[0]
 
-    assert len(output1.outputs[0].token_ids) >= 19
-    assert len(output2.outputs[0].token_ids) < 19
+    assert len(output1.outputs[0].token_ids) >= 10
+    # Logits bias should force eos token appears, then we check if
+    # after min tokens reached the logits processor is properly
+    # cleared.
+    assert len(output1.outputs[0].token_ids) < 20
+    assert len(output2.outputs[0].token_ids) < 10
```
[the remark this answers is not in the document]
Comment (wallashss) [reply tests/e2e/test_sampling_params.py]: NP, PTAL
